pkg/gatherers/conditional/conditional_gatherer.go

```diff
@@ -0,0 +1,380 @@
+func (g *Gatherer) GetName() string {
+	return "conditional"
+}
+
+// GetGatheringFunctions returns gathering functions that should be run considering the conditions
+// + the gathering function producing metadata for the conditional gatherer
+func (g *Gatherer) GetGatheringFunctions(ctx context.Context) (map[string]gatherers.GatheringClosure, error) {
+	err := g.updateAlertsCache(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	gatheringFunctions := make(map[string]gatherers.GatheringClosure)
+
+	gatheringFunctions["conditional_gatherer_rules"] = gatherers.GatheringClosure{
+		Run:     g.GatherConditionalGathererRules,
+		CanFail: canConditionalGathererFail,
+	}
+
+	for _, conditionalGathering := range gatheringRules {
+		allConditionsAreSatisfied, err := g.areAllConditionsSatisfied(conditionalGathering.Conditions)
+		if err != nil {
+			return nil, err
+		}
+
+		if allConditionsAreSatisfied {
+			functions, errs := g.createGatheringClosures(conditionalGathering.GatheringFunctions)
+			if len(errs) > 0 {
+				return nil, err
+			}
+
+			for funcName, function := range functions {
+				gatheringFunctions[funcName] = function
+			}
+		}
+	}
+
+	return gatheringFunctions, nil
+}
+
+// GatherConditionalGathererRules stores the gathering rules in insights-operator/conditional-gatherer-rules.json
+func (g *Gatherer) GatherConditionalGathererRules(context.Context) ([]record.Record, []error) {
+	return []record.Record{
+		{
+			Name: "insights-operator/conditional-gatherer-rules",
+			Item: record.JSONMarshaller{Object: gatheringRules},
+		},
+	}, nil
+}
+
+// ShouldBeProcessedNow returns true when it's time to process this gatherer
+func (g *Gatherer) ShouldBeProcessedNow() bool {
+	return utils.ShouldBeProcessedNow(g.lastProcessingTime, conditionalGathererPeriod)
+}
+
+// UpdateLastProcessingTime updates the time when the gatherer was processed
+func (g *Gatherer) UpdateLastProcessingTime() {
+	g.lastProcessingTime = time.Now()
+}
+
+// areAllConditionsSatisfied returns true if all the conditions are satisfied, for example if the condition is
+// to check if a metric is firing, it will look at that metric and return the result according to that
+func (g *Gatherer) areAllConditionsSatisfied(conditions []conditionWithParams) (bool, error) {
+	for _, condition := range conditions {
+		switch condition.Type {
+		case alertIsFiring:
+			alertName, err := getStringFromMap(condition.Params, "name")
+			if err != nil {
+				return false, err
+			}
+
+			if !g.isAlertFiring(alertName) {
+				return false, nil
+			}
+		default:
+			return false, fmt.Errorf("unknown condition type: %v", condition.Type)
+		}
+	}
+
+	return true, nil
+}
+
+// updateAlertsCache updates the cache with firing alerts
+func (g *Gatherer) updateAlertsCache(ctx context.Context) error {
+	if g.metricsGatherKubeConfig == nil {
+		return nil
+	}
+
+	metricsClient, err := rest.RESTClientFor(g.metricsGatherKubeConfig)
+	if err != nil {
+		return err
```

Comment:
It's failing here with `open /var/run/configmaps/service-ca-bundle/service-ca.crt: no such file or directory` for some reason when running against my quicklab cluster. Looking....

Reply:
Ahh now I see it....we basically ignore this error and just log a warning. See https://github.com/openshift/insights-operator/blob/master/pkg/gatherers/clusterconfig/recent_metrics.go#L42

Reply:
Yeah, the conditional gatherer requires the metrics to be available https://github.com/openshift/insights-operator#debugging-prometheus-metrics-without-valid-ca

Reply:
Can we update it please? I think this is not very convenient (first for running IO locally when you don't have the metrics config). The new conditional gatherer is marked as `canFail=false`, which means that if metrics are not available then IO will go disabled after some time (because this new gatherer will repeatedly fail). So I would suggest one of the following: 

- only log the error here and don't return it (the same we do in our existing metrics gatherer)
- mark the new conditional gatherer as `canFail=true`

Reply:
Hm, yeah, that makes sense, thanks